src/Conversion/InterventionImageFileConverter.php

```diff
@@ -0,0 +1,181 @@
+        }
+        if (!$this->supportedByIntervention($toFormat, $originalBackend)) {
+            throw new FileConversionException("Convertion to format '$toFormat' is not suported.");
+        }
+
+        $quality = $options['quality'] ?? null;
+        // Clone the backend if we're changing quality to avoid affecting other manipulations to that original image
+        $backend = $quality === null ? $originalBackend : clone $originalBackend;
+        // Pass through to invervention image to do the conversion for us.
+        try {
+            $result = $from->manipulateExtension(
+                $toFormat,
+                function (AssetStore $store, string $filename, string $hash, string $variant) use ($backend, $quality) {
+                    if ($quality !== null) {
+                        $backend->setQuality($quality);
+                    }
+                    $config = ['conflict' => AssetStore::CONFLICT_USE_EXISTING];
+                    $tuple = $backend->writeToStore($store, $filename, $hash, $variant, $config);
+                    return [$tuple, $backend];
+                }
+            );
+        } catch (ImageException $e) {
+            throw new FileConversionException('Failed to convert: ' . $e->getMessage(), $e->getCode(), $e);
+        }
+        // This is very unlikely but the API for `manipulateExtension()` allows for it
+        if ($result === null) {
+            throw new FileConversionException('File conversion resulted in null. Check whether original file actually exists.');
+        }
+        return $result;
+    }
+
+    private function validateOptions(array $options): array
+    {
+        $problems = [];
+        foreach ($options as $key => $value) {
+            if ($key !== 'quality') {
+                $problems[] = "unexpected option '$key'";
+                continue;
+            }
+            if (!is_int($value)) {
+                $problems[] = "quality value must be an integer";
+            }
+        }
+        return $problems;
+    }
+
+    private function supportedByIntervention(string $format, InterventionBackend $backend): bool
+    {
+        $driver = $backend->getImageManager()->config['driver'] ?? null;
+        // If the driver is somehow not GD or Imagick, we have no way to know what it might support
+        if ($driver !== 'gd' && $driver !== 'imagick') {
+            return false;
+        }
+
+        // Return early for empty values - we obviously can't support that
+        if ($format === '') {
+            return false;
+        }
+
+        // GD and Imagick support different things.
+        // This follows the logic in intervention's AbstractEncoder::process() method
+        // and the various methods in the Encoder classes for GD and Imagick,
+        // excluding checking for strings that were obviously mimetypes
+        switch (strtolower($format)) {
+            case 'gif':
```

Comment:
Backlog an issue linking to https://github.com/Intervention/image/issues/1326 to refactor this part

Reply:
I'm not going to raise an issue to make a change based on the possibility that intervention image **might** have this functionality in a future release.